### PR TITLE
fix: dump required data in `test-agg` and refactor EVM verifier

### PR DESCRIPTION
**Pull request by silathdiir**
### Summary

1. Dump inner-circuit-proof (struct `TargetCircuitProof` to JSON), agg-circuit-proof (raw-bytes), vk (raw-bytes), instance (raw-bytes), deployment (verifier) yul-code and bytecode.

2. Add a new struct `EvmVerifier` to `verifier.rs`.

### Issue

Close https://github.com/scroll-tech/scroll-zkevm/issues/133

### Test

Test case `test_aggregation_api` is passed, and the dumped files are saved in folder `~/steven/scroll-zkevm/zkevm/output_20230504_082934_multi` of super.

### CODE REVIEW

**Comment by lispc** [comment]
and the solidity verifier

**Review by zhenfeizhang — DISMISSED** [review]

**Comment by lispc** [comment]
nonono, "let deployment_code = prover.create_evm_verifier_bytecode(&agg_circuit, pk_outer.get_vk());" this `deployment_code` is what i need to dump. I think it is YUL bytecode(you have have a look at the create_evm_verifier_bytecode impl details)? and send a copy of this to me,( you can use "super" server to run tests, so i can have a check what it is really yul or something else.

don't import the old "halo2-snark-aggregator". we dont need that

File: zkevm/tests/aggregation_tests.rs [hunk 1]
```diff
@@ -53,41 +63,65 @@ fn test_aggregation_api() {
...
+    let deployment_code = prover.create_evm_verifier_bytecode(&agg_circuit, agg_vk);
     log::info!("finished byte code generation");
 
     // 5. validate the proof with evm bytecode
-    Verifier::evm_verify(deployment_code, instances, proof);
+    Verifier::evm_verify(
+        deployment_code,
+        agg_circuit.instances(),
+        proof.proof.clone(),
+    );
```
Comment (lispc) [inline zkevm/tests/aggregation_tests.rs:112]: need also to dump the instance and proof used by Verifier::evm_verify

**Comment by lispc** [comment]
now i think it will be good to impl https://github.com/scroll-tech/scroll-zkevm/issues/133 in this pr at once

File: zkevm/tests/aggregation_tests.rs (hunk 1 above)
Comment (silathdiir) [reply zkevm/tests/aggregation_tests.rs]: Fix to dump inner-circuit-proof, agg-circuit-proof, vk, instance, and deployment bytecode in commit [478d1d4](https://github.com/scroll-tech/scroll-zkevm/pull/132/commits/478d1d46fe4825b5a91184c6355061251b57c48a).

Will check function `create_evm_verifier_bytecode` impl, then refactor to EVM verifier. Thanks.

**Comment by lispc** [comment]
lgtm.   i am not sure what is deployment_code really is.  soldity or yul or bin .. waiting for the dump

**Comment by silathdiir** [comment]
> lgtm. i am not sure what is deployment_code really is. soldity or yul or bin .. waiting for the dump

Fix to dump yul-code to `verifier.yul` and bytecode to `verifier.bin` in commit [9517860](https://github.com/scroll-tech/scroll-zkevm/pull/132/commits/95178604676f6107c9beec826ca7866963eee04a) and [ab52483](https://github.com/scroll-tech/scroll-zkevm/pull/132/commits/ab52483d74a4d8e58e27cc94a1999d281a8d9942).

**Pull request merged**